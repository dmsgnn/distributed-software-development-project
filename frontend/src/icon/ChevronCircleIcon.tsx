import { Icon, IconProps } from '@chakra-ui/react';
import React, { FC } from 'react';

export const ChevronCircleIcon: FC<IconProps> = (props) => {
	return (
		<Icon color="primary.500" viewBox="0 0 45 45" {...props}>
			<path
				fillRule="evenodd"
				clipRule="evenodd"
				d="M22.5 0a22.5 22.5 0 1 0 0 45 22.5 22.5 0 0 0 0-45Zm-9.25 19.33 6.34 9.5a3.5 3.5 0 0 0 5.82 0l6.34-9.5-2.5-1.66-6.33 9.5a.5.5 0 0 1-.84 0l-6.33-9.5-2.5 1.66Z"
				fill="currentColor"
			/>
		</Icon>
	);
};
