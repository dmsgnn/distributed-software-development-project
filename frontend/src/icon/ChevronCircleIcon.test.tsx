import { axe } from 'jest-axe';

import { act, render } from 'test-utils';
import { ChevronCircleIcon } from './ChevronCircleIcon';

describe('ChevronCircleIcon', () => {
	it('should matches snapshot', () => {
		const { asFragment } = render(<ChevronCircleIcon />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should be accessible', async () => {
		const { container } = render(<ChevronCircleIcon />);

		let results: Awaited<ReturnType<typeof axe>>;
		await act(async () => {
			results = await axe(container);
		});

		expect(results).toHaveNoViolations();
	});
});
