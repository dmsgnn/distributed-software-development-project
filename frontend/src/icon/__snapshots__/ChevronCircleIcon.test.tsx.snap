// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChevronCircleIcon should matches snapshot 1`] = `
<DocumentFragment>
  <svg
    class="chakra-icon css-q5208i"
    focusable="false"
    viewBox="0 0 45 45"
  >
    <path
      clip-rule="evenodd"
      d="M22.5 0a22.5 22.5 0 1 0 0 45 22.5 22.5 0 0 0 0-45Zm-9.25 19.33 6.34 9.5a3.5 3.5 0 0 0 5.82 0l6.34-9.5-2.5-1.66-6.33 9.5a.5.5 0 0 1-.84 0l-6.33-9.5-2.5 1.66Z"
      fill="currentColor"
      fill-rule="evenodd"
    />
  </svg>
</DocumentFragment>
`;
